src/main/java/homework_5/custom_regex_matcher/CustomRegexMatcher.java

```diff
@@ -0,0 +1,34 @@
+package homework_5.custom_regex_matcher;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+class CustomRegexMatcher {
+
+    void run() {
+
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(System.in))) {
+
+            String inputStr = reader.readLine();
+
+            if (inputStr == null) {
+                System.out.println("false");
+                return;
+            }
+
+            String patternStr = "^<([a-z]+)([^>]+)*(?:>(.*)<\\/\\1>|\\s+\\/>)$";
```

Comment:
Opt: could you write to console or somewhere, what is the expected boundaries for input? Example string?